### ISSUE TITLE
deploying to amazon failed

### ISSUE THREAD

**Issue by iuser2** [jlund/streisand#244]
I tried several times. It fails at the same place: "configure the server and install required software." The failure is caused by certain time out. You can see the error message below. 

ansible version: 2.0.0.2
streisand's last commit: [bfee695bf2c572f04d9911ab270cd0dd684cc331](https://github.com/jlund/streisand/commit/bfee695bf2c572f04d9911ab270cd0dd684cc331)

> PLAY [Configure the Server and install required software] **********************
> 
> TASK [setup] *******************************************************************
> fatal: [54.153.11.131]: FAILED! => {"failed": true, "msg": "ERROR! Timeout (12s) waiting for privilege escalation prompt: "}
> 
> NO MORE HOSTS LEFT *************************************************************
> 
> PLAY RECAP *********************************************************************
> 127.0.0.1                  : ok=18   changed=7    unreachable=0    failed=0  
> 54.153.11.131              : ok=0    changed=0    unreachable=0    failed=1

**Comment by jlund** [comment]
Can you SSH to the server that gets created (e.g. `ssh ubuntu@xxx.xxx.xxx.xxx`)?

**Comment by jlund** [comment]
I just created two new Amazon servers using the same Ansible version, so I'm not sure what could be happening here. We should be able to get to the bottom of it with more details.

**Comment by df-sh** [comment]
I was running into this same problem earlier... I am not sure what causes the issue on amazon servers (wasnt happening on rackspace) but the script attempts to continue before the new server is accepting commands.  If you change the ansible config and playbook to target the previously failed server, the script completes.

**Comment by DynaMc** [comment]
Setup fails for me as well on AWS. 
ansible 2.0.1.0

```
PLAY [Configure the Server and install required software] **********************

TASK [setup] *******************************************************************
fatal: [54.172.96.205]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}

PLAY [Open all service ports] **************************************************

TASK [Set the streisand_host fact to the value of the new server. This fact will be used to retrieve the obfsproxy port values.] ***
ok: [127.0.0.1]

TASK [Retrieve the randomly chosen Tor obfs3 port] *****************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'tor_obfs3_port'"}

```

**Comment by jlund** [comment]
After more than ten successful attempts, I was finally able to trigger this. This appears to be similar to an issue that was appearing sporadically with DigitalOcean servers as well. What happens is that the SSH daemon starts running (so the `wait_for` task passes) but the server is too slow to respond and the initial SSH connection fails. It works when you try again because by that point the server is _really_ ready. I added a 90-second pause, which worked well for DigitalOcean too.

**Comment by DynaMc** [comment]
Hmm it still failed for me trying again. No idea why it takes so much longer for my instance to settle but I just tried again and got the same thing. 

Also, as you said the server is up but I have no keys to login. I also tried ./steisand --limit @playbooks/amazon.retry and it was like starting all over again. 

I'll try editing the 90 seconds you added to like 300 and see if it works.

This was the message: 
`TASK [genesis-amazon : New EC2 servers are occasionally slow to process incoming SSH connections even after the OpenSSH daemon has started up. Pause for 90 seconds.] ***
Pausing for 90 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [127.0.0.1]

PLAY [Configure the Server and install required software] **********************

TASK [setup] *******************************************************************
fatal: [54.172.194.142]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}

PLAY [Open all service ports] **************************************************

TASK [Set the streisand_host fact to the value of the new server. This fact will be used to retrieve the obfsproxy port values.] ***
ok: [127.0.0.1]

TASK [Retrieve the randomly chosen Tor obfs3 port] *****************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'tor_obfs3_port'"}

NO MORE HOSTS LEFT *************************************************************
	to retry, use: --limit @playbooks/amazon.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=20   changed=7    unreachable=0    failed=1   
54.172.194.142             : ok=0    changed=0    unreachable=1    failed=0   

`

**Comment by DynaMc** [comment]
Sorry to keep hammering comments out, just trying to provide as much feedback as possible. 

I tried 600 for the timer came back and it still failed for me. Weird. Maybe I should try somewhere other than us-east-1. 

```
Pausing for 600 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [127.0.0.1]

PLAY [Configure the Server and install required software] **********************

TASK [setup] *******************************************************************
fatal: [54.175.148.119]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}

PLAY [Open all service ports] **************************************************

TASK [Set the streisand_host fact to the value of the new server. This fact will be used to retrieve the obfsproxy port values.] ***
ok: [127.0.0.1]

TASK [Retrieve the randomly chosen Tor obfs3 port] *****************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'tor_obfs3_port'"}

NO MORE HOSTS LEFT *************************************************************
	to retry, use: --limit @playbooks/amazon.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=20   changed=7    unreachable=0    failed=1   
54.175.148.119             : ok=0    changed=0    unreachable=1    failed=0   

```

**Comment by jlund** [comment]
@DynaMc I think you might be running into a different issue. Are you able to connect to the servers via SSH outside of Streisand and Ansible (e.g. `ssh ubuntu@ip-address-of-the-new-server`)?

**Comment by aadnow** [comment]
I have the same issue describe. The first few times I tried, I could not SSH into the new instance right away. So I tried several longer pause intervals (below it's at 330sec). It still fails, but I can access via SSH immediately. 

Here's what it looks like:
```
`TASK [genesis-amazon : New EC2 servers are occasionally slow to process incoming SSH connections even after the OpenSSH daemon has started up. Pause for 330 seconds.] ***
Pausing for 330 seconds
(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)
ok: [127.0.0.1]

PLAY [Configure the Server and install required software] **********************

TASK [setup] *******************************************************************
fatal: [54.193.23.80]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}

PLAY [Open all service ports] **************************************************

TASK [Set the streisand_host fact to the value of the new server. This fact will be used to retrieve the obfsproxy port values.] ***
ok: [127.0.0.1]

TASK [Retrieve the randomly chosen Tor obfs3 port] *****************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'tor_obfs3_port'"}

NO MORE HOSTS LEFT *************************************************************
	to retry, use: --limit @playbooks/amazon.retry

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=20   changed=7    unreachable=0    failed=1   
54.193.23.80               : ok=0    changed=0    unreachable=1    failed=0   

ubuntu@ubuntu:~/streisand$ ssh 54.193.23.80
Welcome to Ubuntu 14.04.3 LTS (GNU/Linux 3.13.0-74-generic x86_64)
` 
```
When I run `./streisand --limit @playbooks/amazon.retry` it just re-runs the script as normal and sets up a new instance.
Hope this helps

**Comment by DynaMc** [comment]
Just to note, I can SSH in as well. Nothing appears to be setup though. 

Also, when I run the `./streisand --limit @playbooks/amazon.retry` it runs the setup as normal and starts trying to add a new EC2 instance.

**Comment by aadnow** [comment]
Got it to work but cannot say definitively what fixed it. First try on a fresh Ubuntu install (as opposed to previous live USB) failed. Generated key pair because I couldn't remember if I had done that already. Did it as root. Bingo. Must have generated them as user in the previous attempts? Or could have been the other way around... Anyway: seems like user error in my case. Maybe docs could clarify `ssh-keygen` as user or root.

**Comment by jlund** [comment]
Can anyone who is experiencing this issue (where Ansible seemingly can't connect to configure the server, but a manual SSH works just fine) please post some verbose SSH output (`ssh -v`). I'm wondering if this has something to do with non-RSA keys?

I'm kind of drawing a blank otherwise. I can't duplicate this on any existing or new AWS accounts.

**Comment by victorkwak** [comment]
I just had this issue as well. It turns out it was because I had neglected to copy over the private key into .ssh because I'm an idiot (and I forgot). Yes, I generated the key outside of .ssh and had to copy it in. Anyways, the script checks for the public key, but not the private key, and when you don't have your private key... you know. I don't know if this will help, but I just wanted to share another possible source of this error.
